aten/src/ATen/native/quantized/TensorFactories.cpp

```diff
@@ -94,23 +94,45 @@ Tensor empty_per_channel_affine_quantized_other_backends_stub(
 
 // Create an empty quantized Tensor with size, based on the options
 // and quantization parameters of the input quantized Tensor
-Tensor empty_quantized(IntArrayRef size, const Tensor& qtensor) {
+Tensor empty_quantized(
+    IntArrayRef size,
+    const Tensor& qtensor,
+    c10::optional<ScalarType> dtype,
+    c10::optional<Layout> layout,
+    c10::optional<Device> device,
+    c10::optional<bool> pin_memory,
```

Comment:
It looks like this function allows for quantized tensors to be pinned: Can we extend the test here to test for this? Currently it checks to see 

``` python
def test_cuda_quantization_does_not_pin_memory(self):
        # Context - https://github.com/pytorch/pytorch/issues/41115
        x = torch.randn(3)
        self.assertEqual(x.is_pinned(), False)
        q_int = torch.randint(0, 100, [1, 2, 3], device="cuda", dtype=torch.uint8)
        q = torch._make_per_tensor_quantized_tensor(q_int, scale=0.1, zero_point=0)
        x = torch.randn(3)
        self.assertEqual(x.is_pinned(), False)
```

Reply:
there's a test for pinned tensors, is that sufficient or do we need something else?

If quantized tensors should never be able to be pinned I can just disable that option. As is, this is more of a forward facing change, the only information that changes is device, the rest was just added because the functionality seemed like it might be useful in the future.

The code as is doesn't set the pinned_memory flag (see Tensor_Conversions.cpp function to_impl, pin_out is only used for non_quantized tensors) but it'd be theoretically possible to pin the memory if you called the empty_quantized function directly with that option set.